### ISSUE TITLE
Integrate with TelegramBot

### ISSUE THREAD

**Issue by sleodc** [doctorfree/MMM-YouTubeWebView#1]
Hi, how would you integrate it with telegrambot? 

I am using the Ext-TelegramBot by Bugsonet. Would appreciate any help. 

Thank you

**Comment by doctorfree** [comment]
Hi, `TelegramBot` integration with `YouTubeWebView` is an excellent idea but I believe it would require some enhancements to the module as it currently only acts on the `MODULE_DOM_CREATED` notification. Perhaps another `YouTube` viewer module has `TelegramBot` integration.

Alternately, it might be possible to use a combination of [MirrorCommand](https://github.com/doctorfree/MirrorCommand) and [MMM-TelegramCommands](https://github.com/doctorfree/MMM-TelegramCommands) to accomplish some control of `YouTube` playback via `TelegramBot` and the `mirror` command line utility. It would be an interesting area to explore but I do not believe this is currently supported in any of my modules.

**Comment by sleodc** [comment]
Thanks for the reply. I tried integrating with this (https://github.com/bugsounet/EXT-TelegramBot ), adding this to the code but doesn’t seem to work.

    notificationReceived: function (notification, payload, sender) {
        if (notification == 'MODULE_DOM_CREATED') {
          if (this.config.loadedJS && this.config.loadedJS.length > 0) {
            const webview = document.getElementById('mmm-youtube-webview');
            if (webview) {
              webview.addEventListener('did-finish-load', () => {
                webview.executeJavaScript(this.config.loadedJS);
              });
            } else {
              // TODO: Show Error
            }
          }
        }
    if notification == “EXT_YOUTUBE-PLAY":

      const webview = document.getElementById('mmm-youtube-webview');
        webview.src= "https://www.youtube.com/embed” +payload
    },

EXT_TELBOTCommands: function(commander) {
    commander.add({
      command: "youtube",
      description: this.translate("YouTubeDescription"),
      callback: "tbYoutube"
    })
  },

  tbYoutube: function(command, handler) {
    var query = handler.args
    if (query) {
      var args = query.toLowerCase().split(" ")
      var params = query.split(" ").slice(1).join(" ")
      switch (args[0]) {
        case "play":
          if (params) {
            params = params.split(" ")
            this.notificationReceived("EXT_YOUTUBE-PLAY", params[0])
            handler.reply("TEXT", this.translate("YouTubePlay", { VALUES: params[0] }))
          } else handler.reply("TEXT", "/youtube play <video ID>")
          break
        case "search":
          if (!this.searchInit) return handler.reply("TEXT", this.translate("YouTubeSearchDisabled"))
          if (params) {
            this.notificationReceived("EXT_YOUTUBE-SEARCH", params)
            handler.reply("TEXT", this.translate("YouTubeSearch", { VALUES: params }))
          }
          else handler.reply("TEXT", "/youtube search <youtube title/artist>")
          break
        default:
          handler.reply("TEXT", this.translate("YouTubeCmdNotFound"))
          break
      }
    } else {
    
  }
}
})

**Comment by doctorfree** [comment]
@sleodc yes, notifications are tricky. The `MMM-YouTubeWebView` module was not designed to receive notifications from other modules. In order to do so it would need a `node_helper.js` to handle notifications. To get started learning about inter-module notifications see https://docs.magicmirror.builders/development/core-module-file.html

I am swamped at work right now and do not have time to implement notifications in this module. Perhaps I can get back to this in the future.

**Comment by sleodc** [comment]
ahh ok gotcha, thank you.
